### PR TITLE
Update Gateway Docs to include Mosaic info.

### PR DESCRIPTION
**Pull request by vladimirk-db**
Tested via a local build of docs.

## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -W --keep-going -n -T
```
Comment (vladimirk-db) [inline docs/Makefile]: I'll revert this - but I had to exclude this flag (it's fail-on-warning) to be able to build the docs.

File: docs/source/gateway/index.rst [hunk 2]
```diff
@@ -1194,6 +1194,71 @@ Here are some examples for how you might use curl to interact with the Gateway:
...
+        name="mosaicml-llama-completions",
+        route_type="llm/v1/completions",
+        model={
+            "name": "llama2-70b-chat",
+            "provider": "mosaicml",
+            "mosaicml_config": {
+                "mosaicml_api_key": "<YOUR_MOSAIC_API_KEY>"
+            }
+        }
+    )
```
Comment (dbczumar) [inline docs/source/gateway/index.rst:1226]: Is MosaicML supported for route type `llm/v1/chat` as well? If so, should we provide an example?

File: docs/Makefile (hunk 1 above)
Comment (dbczumar) [reply docs/Makefile]: What's the warning?

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @vladimirk-db !

File: docs/Makefile (hunk 1 above)
Comment (vladimirk-db) [reply docs/Makefile]: "sphinx.errors.SphinxWarning: /Users/vladimir.kolovski/repos/mlflow-vlad/mlflow/mlflow/sklearn/__init__.py:docstring of mlflow.sklearn.autolog:130:duplicate label post training metrics, other instance in /Users/vladimir.kolovski/repos/mlflow-vlad/mlflow/docs/source/python_api/mlflow.pyspark.ml.rst
"

Comment (dbczumar) [reply docs/Makefile]: Got it - thanks!

File: docs/source/gateway/index.rst (hunk 2 above)
Comment (vladimirk-db) [reply docs/source/gateway/index.rst:1226]: Not yet, we'll follow up on it, actually (and update docs then).
